core/src/main/scala/org/apache/spark/ui/UIWorkloadGenerator.scala

```diff
@@ -99,12 +103,17 @@ private[spark] object UIWorkloadGenerator {
             } catch {
               case e: Exception =>
                 println("Job Failed: " + desc)
+            } finally {
+              barrier.release()
             }
           }
         }.start
         Thread.sleep(INTER_JOB_WAIT_MS)
       }
     }
+
+    // Waiting for threads.
+    barrier.acquire()
```

Comment:
It's because I didn't think it's simple to use `join()`; we need to allocate array for those threads and the main thread should call for each thread in the array.

But, I also think it's simpler to use `CounDownLatch` instead of Semaphore right?